### ISSUE TITLE
IndexError: list index out of range

### ISSUE THREAD

**Issue by EldarSehayekZenity** [koxudaxi/datamodel-code-generator#1109]
**Describe the bug**
I executed the parser on a JSON schema file from online source and received "IndexError: list index out of range" error.

**To Reproduce**

Example schema:
I attempted to use the tool on this schema: https://github.com/CycloneDX/specification/blob/master/schema/bom-1.4.schema.json

Used commandline:
```
$ datamodel-codegen  --input .\cyclonedx-1.4.schema.json --input-file-type jsonschema --output cyclonedx.py
```

**Expected behavior**
If there's something wrong in the schema I expect a clear error message, unless there's a bug in the code generator.

**Version:**
 - OS: Windows
 - Python version: 3.10
 - datamodel-code-generator version: 0.17.1

**Additional context**
The error callstack:

UserWarning: format of 'iri-reference' not understood for 'string' - using default
  warn(
Traceback (most recent call last):
  File "C:\Dev\zenitysec\docdb-poc\env\lib\site-packages\datamodel_code_generator\__main__.py", line 687, in main
    generate(
  File "C:\Dev\zenitysec\docdb-poc\env\lib\site-packages\datamodel_code_generator\__init__.py", line 428, in generate
    results = parser.parse()
  File "C:\Dev\zenitysec\docdb-poc\env\lib\site-packages\datamodel_code_generator\parser\base.py", line 947, in parse
    self.parse_raw()
  File "C:\Dev\zenitysec\docdb-poc\env\lib\site-packages\datamodel_code_generator\parser\jsonschema.py", line 1464, in parse_raw
    self._parse_file(self.raw_obj, obj_name, path_parts)
  File "C:\Dev\zenitysec\docdb-poc\env\lib\site-packages\datamodel_code_generator\parser\jsonschema.py", line 1555, in _parse_file
    self.parse_raw_obj(key, model, path)
  File "C:\Dev\zenitysec\docdb-poc\env\lib\site-packages\datamodel_code_generator\parser\jsonschema.py", line 1403, in parse_raw_obj
    self.parse_obj(name, JsonSchemaObject.parse_obj(raw), path)
  File "C:\Dev\zenitysec\docdb-poc\env\lib\site-packages\datamodel_code_generator\parser\jsonschema.py", line 1416, in parse_obj
    self.parse_root_type(name, obj, path)
  File "C:\Dev\zenitysec\docdb-poc\env\lib\site-packages\datamodel_code_generator\parser\jsonschema.py", line 1095, in parse_root_type
    data_type = data_types[0]
IndexError: list index out of range

**Comment by phumphre** [comment]
I get the same error using a publicly available (and valid) jsonschema (from NCBI).

**Comment by koxudaxi** [comment]
Thank you for posting the issue.
I will fix it.

**Comment by malte-behrendt-gr** [comment]
Thanks for this awesome library!

I have the same issue with https://json.schemastore.org/sarif-2.1.0-rtm.5.json

**Comment by howardj99** [comment]
In case it helps, I’m able to parse the [CycloneDX schema](https://github.com/CycloneDX/specification/blob/master/schema/bom-1.4.schema.json)  using `datamodel-code-generator` version 0.15.0 at the latest, so I think this issue may have been originally introduced with the release of 0.16.0

**Comment by koxudaxi** [comment]
@EldarSehayekZenity @phumphre @malte-behrendt-gr @howardj99 
I have released the new version, `0.17.2`.
Thank you very much!!
